File: server/router/ExpertDatabase/expert_tag.go
package ExpertDatabase

import (
	v1 "github.com/flipped-aurora/gin-vue-admin/server/api/v1"
	"github.com/flipped-aurora/gin-vue-admin/server/middleware"
	"github.com/gin-gonic/gin"
)

type ExpertTagRouter struct{}

// InitExpertTagRouter 初始化 专家标签 路由信息
func (s *ExpertTagRouter) InitExpertTagRouter(Router *gin.RouterGroup) {
	expertTagRouter := Router.Group("expertTag").Use(middleware.OperationRecord())
	expertTagRouterWithoutRecord := Router.Group("expertTag")
	expertTagApi := v1.ApiGroupApp.ExpertDatabaseApiGroup.ExpertTagApi
	{
		expertTagRouter.POST("createExpertTag", expertTagApi.CreateExpertTag)             // 新建标签
		expertTagRouter.DELETE("deleteExpertTag", expertTagApi.DeleteExpertTag)           // 删除标签
		expertTagRouter.PUT("updateExpertTag", expertTagApi.UpdateExpertTag)              // 更新标签
		expertTagRouter.POST("setExpertTagRelations", expertTagApi.SetExpertTagRelations) // 设置专家标签关联
	}
	{
		expertTagRouterWithoutRecord.GET("findExpertTag", expertTagApi.FindExpertTag)                     // 根据ID获取标签
		expertTagRouterWithoutRecord.GET("getExpertTagList", expertTagApi.GetExpertTagList)               // 分页获取标签列表
		expertTagRouterWithoutRecord.GET("getExpertTagsByExpertId", expertTagApi.GetExpertTagsByExpertId) // 获取某专家关联的标签
	}
}

File: web/src/api/expertTag.js
import service from '@/utils/request'

// @Tags ExpertTag
// @Summary 创建标签
// @Router /expertTag/createExpertTag [post]
export const createExpertTag = (data) => {
  return service({
    url: '/expertTag/createExpertTag',
    method: 'post',
    data
  })
}

// @Tags ExpertTag
// @Summary 删除标签
// @Router /expertTag/deleteExpertTag [delete]
export const deleteExpertTag = (params) => {
  return service({
    url: '/expertTag/deleteExpertTag',
    method: 'delete',
    params
  })
}

// @Tags ExpertTag
// @Summary 更新标签
// @Router /expertTag/updateExpertTag [put]
export const updateExpertTag = (data) => {
  return service({
    url: '/expertTag/updateExpertTag',
    method: 'put',
    data
  })
}

// @Tags ExpertTag
// @Summary 用id查询标签
// @Router /expertTag/findExpertTag [get]
export const findExpertTag = (params) => {
  return service({
    url: '/expertTag/findExpertTag',
    method: 'get',
    params
  })
}

// @Tags ExpertTag
// @Summary 分页获取标签列表
// @Router /expertTag/getExpertTagList [get]
export const getExpertTagList = (params) => {
  return service({
    url: '/expertTag/getExpertTagList',
    method: 'get',
    params
  })
}

// @Tags ExpertTag
// @Summary 设置专家标签关联
// @Router /expertTag/setExpertTagRelations [post]
export const setExpertTagRelations = (data) => {
  return service({
    url: '/expertTag/setExpertTagRelations',
    method: 'post',
    data
  })
}

// @Tags ExpertTag
// @Summary 获取某专家关联的标签
// @Router /expertTag/getExpertTagsByExpertId [get]
export const getExpertTagsByExpertId = (params) => {
  return service({
    url: '/expertTag/getExpertTagsByExpertId',
    method: 'get',
    params
  })
}
